### ISSUE TITLE
🐛 Import Sorting should be case-insensitive

### ISSUE THREAD

**Issue by axetroy** [biomejs/biome#4009]
### Environment information

```block
CLI:
  Version:                      1.9.2
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           "v16.20.2"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

Ideally, Import Sorting should be case-insensitive.

Because the file system explorer (and all editor) does the same thing.

Input:

```js
import Common from 'common'
import Link from 'Link'
```

Result:

```js
import Link from 'Link'
import Common from 'common'
```

### Expected result

Import Sorting should be case-insensitive

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
Closing because making import sorting case-insensitive is part of the RFC. See #3177
